tests/accelerators/test_tpu.py

```diff
@@ -332,3 +333,15 @@ def test_devices_auto_choice_tpu():
     trainer = Trainer(accelerator="auto", devices="auto")
     assert trainer.devices == 8
     assert trainer.tpu_cores == 8
+
+
+@RunIf(tpu=True)
+def test_teardown(tmpdir):
+    model = BoringModel()
+    strategy = TPUSpawnStrategy()
+    with patch.object(TPUSpawnStrategy, "teardown", wraps=strategy.teardown) as mock:
+        trainer = Trainer(strategy=strategy, max_epochs=1, tpu_cores=8, default_root_dir=tmpdir, fast_dev_run=True)
```

Comment:
redundant with fast_dev_run
```suggestion
        trainer = Trainer(strategy=strategy, tpu_cores=8, default_root_dir=tmpdir, fast_dev_run=True)
```